### PR TITLE
APS 827: Remove legacy actions from workflow manager role

### PR DESCRIPTION
**Pull request by edavey**
Jira [APS-827: Reduce scope of interim "v1 Manage" section](https://dsdmoj.atlassian.net/browse/APS-827)

We'll be moving to a new role (suggested `CRU_TEAM`) for actions which live
within the "Manage" section but aren't for AP Staff (`*MANAGER` roles).

See this table for the roles which are now shown each of the **booking actions**:

```
| Action             | Roles                   | notes                                                   |
| ------------------ | ----------------------- | ------------------------------------------------------- |
| move person        | LEGACY_MANAGER, MANAGER | NE users of "v1 manage"                                 |
| change dates       | LEGACY_MANAGER, MANAGER | NE users of "v1 manage"                                 |
| arrived            | MANAGER                 | deprecated feature, will return improved in "v2 manage" |
| not arrived        | MANAGER                 | deprecated feature, will return improved in "v2 manage" |
| departed           | MANAGER                 | deprecated feature, will return improved in "v2 manage" |
| withdraw placement | WORKFLOW_MANAGER        | we'll switch to CRU_TEAM                                |
```

and the following **premises actions**:

```

| Action             | Roles            | notes                                                   |
| ------------------ | ---------------- | ------------------------------------------------------- |
| withdraw placement | WORKFLOW_MANAGER | we'll switch to CRU_TEAM                                |
| create placement   | WORKFLOW_MANAGER | we'll switch to CRU_TEAM                                |
| view calendar      | MANAGER          | deprecated feature, will return improved in "v2 manage" |

```

We'll re-assign roles so that:

- NE CRU staff get the `LEGACY_MANAGER` role
- NE CRU staff lose the `MANAGER` role to remove deprecated features
- CRU staff continues to use `WORKFLOW_MANAGER` to access  "manage" features which aren't for AP staff. Very soon we'll introduce the new `CRU_TEAM` role which we'll use to control access to these features as well as to the new "CRU staff view all out-of-service-beds".
- internal dev team members can continue to review the deprecated features via the `MANAGER` role

### Tasks

- [x] Refactor the `premisesActions` and `bookingActions` and their tests, and apply the update rules as per the tables above
- [x] Review and adapt the integration tests so that they are run by a user with a Role (or multiple roles) to perform the happy paths
- [ ] Review the E2E tests and ensure that the E2E user has the necessary roles

### CODE REVIEW

File: server/utils/bookings/index.test.ts [hunk 1]
```diff
@@ -303,7 +303,7 @@ describe('bookingUtils', () => {
     it('should return a departure action if a booking is arrived', () => {
       const booking = bookingFactory.arrived().build()
 
-      expect(legacyBookingActions(booking)).toEqual([
+      expect(originalBookingActions(booking)).toEqual([
```
Comment (richpjames) [inline server/utils/bookings/index.test.ts]: Is this in the wrong commit?

**Comment by edavey** [comment]
@richpjames I think this is now ready to merge, if you would be good enough to take another look?

File: server/utils/bookings/bookingActionsByRole.test.ts [hunk 2]
```diff
@@ -0,0 +1,208 @@
+import { bookingActions } from './bookingActions'
+import { bookingFactory, userDetailsFactory } from '../../testutils/factories'
+import applyPaths from '../../paths/apply'
+import paths from '../../paths/manage'
+
+describe('bookingUtils bookingActions by role', () => {
+  const OLD_ENV = process.env
+
+  beforeEach(() => {
```
Comment (richpjames) [inline server/utils/bookings/bookingActionsByRole.test.ts:9]: Again, not sure this is neccessary

**Review by richpjames — COMMENTED** [review]
On the whole this looks great, just a couple of v minor things

File: server/utils/bookings/bookingActions.test.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import { bookingActions } from './bookingActions'
+import { bookingFactory, userDetailsFactory } from '../../testutils/factories'
+import applyPaths from '../../paths/apply'
+
+describe('bookingUtils bookingActions', () => {
+  const OLD_ENV = process.env
```
[the remark this answers is not in the document]
Comment (edavey) [reply server/utils/bookings/bookingActions.test.ts:7]: ah, thanks, that is copy-paste cruft! thanks

File: server/utils/bookings/index.test.ts (hunk 1 above)
Comment (edavey) [reply server/utils/bookings/index.test.ts]: yes, thanks for spotting this

**Comment by edavey** [comment]
@richpjames I've now confirmed that the E2E tests are all passing so if you are happy to approve now I think we can merge this...

**Review by richpjames — APPROVED** [review]

**Pull request merged**